Assets/BoomerangAttack.cs

```diff
@@ -0,0 +1,55 @@
+using UnityEngine;
+using System.Collections;
+
+public class BoomerangAttack : MonoBehaviour
+{
+    [SerializeField] private float speed = 20f;
+    [SerializeField] private float attackRate = 1f;
+    [SerializeField] private float maxDistance = 20f;
+
+    private float nextAttackTime = 0f;
+
+    void Update()
+    {
+        if (Input.GetMouseButtonDown(0) && Time.time >= nextAttackTime)
```

Comment:
We should get an input for an "Attack" rather than a mouse button down to make this compatible with controller.